logging/slog_logger.go

```diff
@@ -1,75 +1,112 @@
 package logging
 
 import (
+	"context"
 	"fmt"
+	"io"
 	"log/slog"
 	"os"
+	"runtime"
+	"time"
 )
 
 type SLogger struct {
-	logger *slog.Logger
+	*slog.Logger
 }
 
 var _ Logger = (*SLogger)(nil)
 
-func NewSlogLogger(env LogLevel) Logger {
-	if env == Production {
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else if env == Development {
-		// Even though code is same for both environments, we are keeping it separate to show that we can have different
-		// implementations for different environments. For example, we can have log levels set differently
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else {
-		panic(fmt.Sprintf("Unknown environment. Expected %s or %s. Received %s.", Development, Production, env))
+// NewSlogTextLogger creates a new SLogger with a text handler
+//
+//	outputWriter is the writer to write the logs to (typically os.Stdout,
+//	  but can also use a io.MultiWriter(os.Stdout, file) to write to multiple outputs)
+//	logLevel is the minimum level to log
+//	logSource if true, adds source information to the log
+func NewSlogTextLogger(outputWriter io.Writer, logLevel slog.Level, logSource bool) *SLogger {
+	handler := slog.NewTextHandler(outputWriter, &slog.HandlerOptions{
+		Level:     logLevel, // prints debug and above
```

Comment:
nit: `// prints this level and above`

Reply:
[78e783b](https://github.com/Layr-Labs/eigensdk-go/pull/127/commits/78e783be8562d48f38ca5483ea7f3e5fd5d1a53b)